NextcloudTalk/NewRoomTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+            }
+        }
+
+        // Fix uisearchcontroller animation
+        self.extendedLayoutIncludesOpaqueBars = true
+
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithOpaqueBackground()
+        appearance.titleTextAttributes = [.foregroundColor: NCAppBranding.themeTextColor()]
+        appearance.backgroundColor = NCAppBranding.themeColor()
+        self.navigationItem.standardAppearance = appearance
+        self.navigationItem.compactAppearance = appearance
+        self.navigationItem.scrollEdgeAppearance = appearance
+
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(self.cancelButtonPressed))
+        self.navigationItem.leftBarButtonItem?.tintColor = NCAppBranding.themeTextColor()
+
+        self.tableView.register(UINib(nibName: kContactsTableCellNibName, bundle: nil), forCellReuseIdentifier: kContactCellIdentifier)
+    }
+
+    func cancelButtonPressed() {
+        self.dismiss(animated: true, completion: nil)
+    }
+
+    // MARK: - New room options
+
+    func getNewRoomOptions() -> [NewRoomOption] {
+        var options = [NewRoomOption]()
+
+        // New group room
+        if NCSettingsController.sharedInstance().canCreateGroupAndPublicRooms() {
+            options.append(.createNewRoom)
+        }
+
+        // List open rooms
+        if NCDatabaseManager.sharedInstance().serverHasTalkCapability(kCapabilityListableRooms, forAccountId: self.account.accountId) {
+            options.append(.joinOpenRoom)
+        }
+
+        return options
+    }
+
+    // MARK: - Contacts
+
+    func getPossibleContacts() {
+        NCAPIController.sharedInstance().getContactsFor(account, forRoom: "new", groupRoom: false, withSearchParam: "") { indexes, _, contactList, error in
+            if error == nil, let indexes = indexes as? [String], let contactList = contactList as? [NCUser] {
+                let storedContacts = NCContact.contacts(forAccountId: self.account.accountId, contains: nil)
+                let combinedContactList = NCUser.combineUsersArray(storedContacts, withUsersArray: contactList)
+                if let combinedContacts = NCUser.indexedUsers(fromUsersArray: combinedContactList) {
+                    let combinedIndexes = Array(combinedContacts.keys).sorted { $0.localizedCaseInsensitiveCompare($1) == .orderedAscending }
+                    self.indexes.append(contentsOf: combinedIndexes)
+                    self.contacts = combinedContacts
+                    self.tableView.reloadData()
+                }
+            }
+        }
+    }
+
+    // MARK: - Search
+
+    func updateSearchResults(for searchController: UISearchController) {
+        self.resultTableViewController.showSearchingUI()
+        DispatchQueue.main.async {
+            self.searchTimer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(self.searchForContacts), userInfo: nil, repeats: false)
```

Comment:
I think there's a missing invalidation here. Shouldn't the search timer be invalidated at some point to have a correct debounce?

Reply:
I was pretty sure I added the timer invalidation here :P